src/SearchImpl/Indexing/CentralizedIndexingActivityQueue.cs

```diff
@@ -0,0 +1,291 @@
+﻿using SenseNet.ContentRepository.Search;
+using SenseNet.ContentRepository.Storage;
+using SenseNet.ContentRepository.Storage.Data;
+using SenseNet.Diagnostics;
+using SenseNet.Search.Indexing.Activities;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SenseNet.Search.Indexing
+{
+    internal static class CentralizedIndexingActivityQueue
+    {
+        private static readonly int MaxCount = 10;
+        private static readonly int RunningTimeoutInSeconds = Configuration.Indexing.LuceneActivityTimeoutInSeconds;
+        private static readonly int LockRefreshPeriodInMilliseconds = RunningTimeoutInSeconds * 3000 / 4;
+        private static readonly int HearthBeatMilliseconds = 1000;
+
+        private static readonly TimeSpan _waitingPollingPeriod = TimeSpan.FromSeconds(2);
+        private static readonly TimeSpan _healthCheckPeriod = TimeSpan.FromMinutes(2);
+        private static readonly TimeSpan _deleteFinishedPeriod = TimeSpan.FromMinutes(23);
+        private static readonly int _activeTaskLimit = 43;
```

Comment:
How this magic number came to the tasks as a limit? I cannot see why it is the right amount of tasks we can activate as a maximum.

Reply:
Is this a wrong value? Is the 100 better?